packages/lesswrong/components/comments/CommentsItem/CommentsItem.tsx

```diff
@@ -148,8 +153,83 @@ const styles = (theme: ThemeType): JssStyles => ({
   excerpt: {
     marginBottom: 8,
   },
+  doppelCommentsButton: {
+    "&.MuiButton-root": {
+      backgroundColor: theme.palette.greyAlpha(0.1),
+      padding: theme.spacing.unit,
+      paddingLeft: theme.spacing.unit*3,
+      paddingRight: theme.spacing.unit*3,
+      borderRadius: theme.borderRadius.small,
+      textTransform: "none",
+      fontVariant: "small-caps",
+      fontColor: theme.palette.text.primary,
+      marginRight: theme.spacing.unit,
+      fontSize: theme.baseFontSize,
+    }
+  },
+  doppelCommentsOptOutButton: {
+    "&&": {
+      backgroundColor: theme.palette.greyAlpha(0),
+    }
+  },
+  adminTriedToOptOutMessage: {
+    color: theme.palette.error.main,
+  }
 });
 
+type PotentialCommentBody = { content: string|null, doppelId: string|null }
+
+const DoppelCommentVote = ({ className, comment, createDoppelVote, potentialCommentBodies, activePotentialComment }:
+  {
+    className: string,
+    comment: CommentsList,
+    createDoppelVote: WithCreateFunction<"DoppelCommentVotes">,
+    potentialCommentBodies: PotentialCommentBody[],
+    activePotentialComment: 0|1|2,
+  }) => {
+  const [adminTriedToOptOut, setAdminTriedToOptOut] = React.useState<string | null>(null)
+  const { LWTooltip } = Components
+  const classes = useStyles(styles, "CommentBody")
+  const currentUser = useCurrentUser()
+  const { flash } = useMessages()
+
+  const voteForThisDoppel = (type: "skip" | "vote", doppelCommentChoiceId?: string) => {
+    if (!currentUser) return
+    void createDoppelVote({
+      data: {
+        userId: currentUser._id,
+        commentId: comment._id,
+        type,
+        doppelCommentChoiceId,
+      },
+    })
+    if (type === "vote") {
+      const message = doppelCommentChoiceId === null ? "You were correct" : "You were incorrect"
```

Comment:
All the information to figure out whether a comment is a doppel comment is available client-side to users?